### ISSUE TITLE
Volumes of type `tmpfs` lacks support for the `mode` parameter

### ISSUE THREAD

**Issue by rawtaz** [docker/compose#9873]
Docker Compose currently does not support the `mode` parameter for `tmpfs` mounts. It only support the `size` parameter.

This has been verified both empirically with `Docker Compose version v2.6.0` with `Docker Engine - Community version 20.10.17` and by looking in the source code.

Here are some references showing that `mode` is currently not supported:
- The `buildTmpfsOptions` function is [missing it and even has a comment for it](https://github.com/docker/compose/blob/2977f4c89740809f20096673ab4e3ced543a905a/pkg/compose/create.go#L1026).
- The `ServiceVolumeTmpfs` struct is [missing it](https://github.com/compose-spec/compose-go/blob/5135b47ac4bb0c9b5c5bcb274e767b410ecef42e/types/types.go#L830).

Worth noting is that Docker does support it in the [`mount.TmpfsOptions` struct](https://github.com/moby/moby/blob/master/api/types/mount/mount.go#L115) that is [returned by the `buildTmpfsOptions` function](https://github.com/docker/compose/blob/2977f4c89740809f20096673ab4e3ced543a905a/pkg/compose/create.go#L1020) in Compose.

Also worth mentioning is that the official Docker Compose specification documentation for tmpfs volumes [clearly states that `mode` should be supported for `tmpfs`](https://docs.docker.com/compose/compose-file/#long-syntax-4). Despite this, however, [the `compose-go` library does not support the `mode` parameter](https://github.com/compose-spec/compose-go/issues/302).

Here's an example of a `tmpfs` specification for a service that does not work (Compose throws an error `services.foo.volumes.0.tmpfs Additional property mode is not allowed`):
~~~
volumes:
  - type: tmpfs
    target: /test
    tmpfs:
      size: 128k
      mode: 0777
~~~

I'm not sure if I got all of the relationships between specifications, libraries and other software right, I figured that given the above mentioned pieces of `mode` throughout the source code, it made sense to open an issue about it in this repository.

It looks like it's rather straight forward to add support for this option, since Docker itself supports it, or am I missing something?

Either way, it's a problem because it's currently impossible to mount and use `tmpfs` volumes with unprivileged processes/containers.

**Comment by cavo789** [comment]
Same issue

I just met the same issue. I'm building a postgres image where I'm using a non root user and tmpfs storage for my PGDATA directory.

The error message I got is `mkdir: can't create directory '/var/lib/postgresql/data': Permission denied` and this is because I'm using a volume like this:

```yml
volumes:
      - type: tmpfs
        target: /var/lib/postgresql/data
        tmpfs:
          size: 536870912 # 512 MB
```

I don't have that error with this volume:

```yml
volumes:
      - my_postgres_data:/var/lib/postgresql/data:delegated
```

So linked to `tmpfs`.

After many searches, I've found some docs (like https://docs.docker.com/storage/tmpfs/#specify-tmpfs-options) suggesting to set the mode to `1770`.

According this doc (https://github.com/compose-spec/compose-spec/blob/master/spec.md#long-syntax-4); I should be able to write the portion below in my `docker-compose.yml` file but `docker compose config` mention it's invalid.  *(The error message is then `services.my_postgres.volumes.0.tmpfs Additional property mode is not allowed`.)*

```yml
volumes:
      - type: tmpfs
        target: /var/lib/postgresql/data
        tmpfs:
          mode: 1700
          size: 536870912 # 512 MB
```

My editor (vscode) is not protesting agains the schema (https://raw.githubusercontent.com/compose-spec/compose-spec/master/schema/compose-spec.json); seems well valid.
